torchgeo/datamodules/deepglobelandcover.py

```diff
@@ -59,8 +60,9 @@ def setup(self, stage: str) -> None:
         """
         if stage in ["fit", "validate"]:
             self.dataset = DeepGlobeLandCover(split="train", **self.kwargs)
-            self.train_dataset, self.val_dataset = dataset_split(
-                self.dataset, self.val_split_pct
+            generator = torch.Generator().manual_seed(0)
```

Comment:
Should we be setting the random seed manually? Seems like this would conflict if a user is already setting a global random seed on purpose to evaluate random runs

Reply:
In the case a user runs `torchgeo fit` followed by `torchgeo validate`, we need to ensure that the same split is chosen both times. This is even more important for datamodules like NASA Marine Debris that choose train/val/test all at once. That's not to say that we can't or shouldn't add a `seed` parameter to all datamodules. I would be open to a PR that does this (so far this PR just preserves the current behavior).

Reply:
Ahh that's right we were already setting the seed previously. Carry on